packages/strapi-plugin-i18n/admin/src/index.js

```diff
@@ -1,28 +1,68 @@
+import * as yup from 'yup';
 import pluginPkg from '../../package.json';
+import middlewares from './middlewares';
 import pluginId from './pluginId';
 import pluginLogo from './assets/images/logo.svg';
 import trads from './translations';
 
 export default strapi => {
   const pluginDescription = pluginPkg.strapi.description || pluginPkg.description;
 
+  middlewares.forEach(middleware => {
+    strapi.middlewares.add(middleware);
+  });
+
   const plugin = {
-    blockerComponent: null,
-    blockerComponentProps: {},
     description: pluginDescription,
     icon: pluginPkg.strapi.icon,
     id: pluginId,
     isReady: true,
-    initializer: () => null,
-    injectedComponents: [],
     isRequired: pluginPkg.strapi.required || false,
-    layout: null,
-    lifecycles: () => {},
     mainComponent: null,
     name: pluginPkg.strapi.name,
     pluginLogo,
-    preventComponentRendering: false,
     trads,
+    boot(app) {
+      const ctbPlugin = app.getPlugin('content-type-builder');
+      const ctbForms = ctbPlugin.internals.forms;
+
+      if (ctbPlugin) {
+        ctbForms.components.add({ id: 'localesPicker', component: () => 'locale picker' });
+
+        ctbForms.extendContentType({
+          validator: {
+            i18n: yup.bool().required(),
+          },
+          form: {
+            advanced() {
+              return [
+                [{ name: 'i18n', type: 'checkbox', label: { id: 'i18nTest' } }],
+                [
+                  {
+                    name: 'i18n-locales',
+                    type: 'localesPicker',
+                    label: { id: 'Select i18n locales' },
+                  },
+                ],
+              ];
+            },
+          },
+        });
+
+        ctbForms.extendFields(['text', 'string'], {
```

Comment:
I'm lacking context here 😅 . Could you explain to me the first argument `['text', 'string']` please ?

Reply:
The first argument is the array of the kind of attributes a plugin wants to extend the idea is to let plugin extend specific types of attributes instead of all of them.